### PR TITLE
Add Picrust2

### PR DESCRIPTION
**Pull request by d4straub**
This solves the old issue #41 

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](<https://github.com/>nf-core/ampliseq/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/ampliseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CITATIONS.md [hunk 1]
```diff
@@ -60,6 +60,15 @@
 * [ITSx](https://besjournals.onlinelibrary.wiley.com/doi/10.1111/2041-210X.12073)
     > Bengtsson-Palme, J., Ryberg, M., Hartmann, M., Branco, S., Wang, Z., Godhe, A., De Wit, P., Sánchez-García, M., Ebersberger, I., de Sousa, F., Amend, A., Jumpponen, A., Unterseher, M., Kristiansson, E., Abarenkov, K., Bertrand, Y.J.K., Sanli, K., Eriksson, K.M., Vik, U., Veldre, V. and Nilsson, R.H.. Improved software detection and extraction of ITS1 and ITS2 from ribosomal ITS sequences of fungi and other eukaryotes for analysis of environmental sequencing data. Methods Ecol Evol 2013, 4: 914-919. doi: 10.1111/2041-210X.12073.
 
+* [PICRUSt2](https://pubmed.ncbi.nlm.nih.gov/32483366/)
+    > Douglas GM, Maffei VJ, Zaneveld JR, Yurgel SN, Brown JR, Taylor CM, Huttenhower C, Langille MGI. PICRUSt2 for prediction of metagenome functions. Nat Biotechnol. 2020 Jun;38(6):685-688. doi: 10.1038/s41587-020-0548-6. PMID: 32483366; PMCID: PMC7365738.
+
+* PICRUSt2 is using in this pipeline by default [EPA-ng](https://pubmed.ncbi.nlm.nih.gov/30165689/)
```
Comment (erikrikarddaniel) [inline CITATIONS.md:66]: Grammar is not correct in "is using in", do you mean "is used in" or maybe "is using"? If the latter, maybe "pipeline" is not correct, maybe just "PICRUSt2 is by default using [EPA-ng]"?

File: CITATIONS.md [hunk 2]
```diff
@@ -60,6 +60,15 @@
...
+    > Barbera P, Kozlov AM, Czech L, Morel B, Darriba D, Flouri T, Stamatakis A. EPA-ng: Massively Parallel Evolutionary Placement of Genetic Sequences. Syst Biol. 2019 Mar 1;68(2):365-369. doi: 10.1093/sysbio/syy054. PMID: 30165689; PMCID: PMC6368480.
+
+* PICRUSt2 is using in this pipeline by default [MinPath](https://pubmed.ncbi.nlm.nih.gov/19680427/)
```
Comment (erikrikarddaniel) [inline CITATIONS.md:69]: Same as above.

File: docs/output.md [hunk 3]
```diff
@@ -27,6 +27,7 @@ The pipeline is built using [Nextflow](https://www.nextflow.io/) and processes d
                 * [Alpha diversity indices](#alpha-diversity-indices) - Diversity within samples
                 * [Beta diversity indices](#beta-diversity-indices) - Diversity between samples (e.g. PCoA plots)
             * [ANCOM](#ancom) - Differential abundance analysis
+        * [PICRUSt2](#picrust2) - Predicting functional abundances
```
Comment (erikrikarddaniel) [inline docs/output.md:30]: Or "Predicting functional genes", "Predicting functional gene abundances"?

File: nextflow_schema.json [hunk 4]
```diff
@@ -103,6 +103,10 @@
                     "description": "Naming of sequencing files",
                     "help_text": "Indicates the naming of sequencing files (default: `\"/*_R{1,2}_001.fastq.gz\"`).\n\nPlease note:\n\n1. The prepended slash (`/`) is required\n2. The star (`*`) is the required wildcard for sample names\n3. The curly brackets (`{}`) enclose the orientation for paired end reads, seperated by a comma (`,`).\n4. The pattern must be enclosed in quotes\n\nFor example for one sample (name: `1`) with forward (file: `1_a.fastq.gz`) and reverse (file: `1_b.fastq.gz`) reads in folder `data`:\n\n```bash\n--input \"data\" --extension \"/*_{a,b}.fastq.gz\"\n```"
                 },
+                "picrust": {
+                    "type": "boolean",
+                    "description": "If functional abundances are predicted from the marker genes."
```
Comment (erikrikarddaniel) [inline nextflow_schema.json:108]: "Functional gene abundances predicted from phylogenetic relationships to marker gene organisms."?

File: workflows/ampliseq.nf [hunk 5]
```diff
@@ -520,6 +521,18 @@ workflow AMPLISEQ {
         }
     }
 
+    //
+    // MODULE: Predict functional abundances from marker genes with Picrust2
```
Comment (erikrikarddaniel) [inline workflows/ampliseq.nf:525]: "functional gene abundances"?

**Review by erikrikarddaniel — APPROVED** [review]
Nice work, just some grammar suggestions.

File: CITATIONS.md (hunk 1 above)
Comment (d4straub) [reply CITATIONS.md:66]: Thanks, will change!

File: CITATIONS.md (hunk 2 above)
Comment (d4straub) [reply CITATIONS.md:69]: Thanks, will change!

File: docs/output.md (hunk 3 above)
Comment (d4straub) [reply docs/output.md:30]: I changed to "Predict the functional potential of a bacterial community"

File: workflows/ampliseq.nf (hunk 5 above)
Comment (d4straub) [reply workflows/ampliseq.nf:525]: Changed.

File: nextflow_schema.json (hunk 4 above)
Comment (d4straub) [reply nextflow_schema.json:108]: Changed.

**Comment by d4straub** [comment]
The tests fail most likely because Picrust2 needs more than the 6GB RAM that is available for ci tests. It seems that the required RAM is independent of the ASV numbers but might be due to the databases that need to be loaded. I'll keep it in the `test_full` profile which is executed in AWS on demand/release, but not in `test_doubleprimers`.

**Comment by erikrikarddaniel** [comment]
Den tors 29 juli 2021 kl 09:11 skrev Daniel Straub ***@***.***
>:

> The tests fail most likely because Picrust2 needs more than the 6GB RAM
> that is available for ci tests. It seems that the required RAM is
> independent of the ASV numbers but might be due to the databases that need
> to be loaded. I'll keep it in the test_full profile which is executed in
> AWS on demand/release, but not in test_doubleprimers.
>
Probably the phylogenetic placement. The reference tree is just as large no
matter the number of ASVs.

**Pull request merged**